torch/nn/utils/parametrizations.py

```diff
@@ -86,11 +97,15 @@ def _update_vectors(self, weight_mat: torch.Tensor) -> None:
         self.v = self.v.clone(memory_format=torch.contiguous_format)
 
     def forward(self, weight: torch.Tensor) -> torch.Tensor:
-        weight_mat = self._reshape_weight_to_matrix(weight)
-        if self.training:
-            self._update_vectors(weight_mat)
-        sigma = torch.dot(self.u, torch.mv(weight_mat, self.v))
-        return weight / sigma
+        if weight.ndim == 1:
```

Comment:
Is there a use case for the `ndim = 1` case? Wondering if its more likely that someone registering spectral_norm for vector is a user error - in that case, might be better to just have an explicit error instead of silently doing something unexpected.

Reply:
Alternatively: just add a note in the doc mentioning behavior on 1D

Reply:
I'll add the note